engine/CMakeLists.txt

```diff
@@ -503,7 +503,7 @@ ENDIF ()
 
 # Set the default build type to Release
 IF (NOT CMAKE_BUILD_TYPE)
-    SET(CMAKE_BUILD_TYPE RelWithDebInfo CACHE STRING "Release, Debug, Profiler" FORCE )
```

Comment:
We need `RelWithDebInfo` so that if someone files a bug we have some debug information to get.
Basically build types:

- Release -> RelWithDebInfo
- Debug -> Debug
- Profile -> Debug???

Reply:
I've been building the "Release" target all along, and that seems to work fine. 
Building with "Debug" also seems to work fine. 

When specifying "RelWithDebInfo" I get a compiler error, right at the start:

[  0%] Building CXX object CMakeFiles/OPcollide.dir/src/cmd/collide2/Ice/IceAABB.cpp.o
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/10.1.0/include/stdint.h:9,
                 from /usr/include/c++/10.1.0/bits/atomic_base.h:36,
                 from /usr/include/c++/10.1.0/bits/shared_ptr_atomic.h:33,
                 from /usr/include/c++/10.1.0/memory:85,
                 from /data/OS/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/universe_generic.h:27,
                 from /data/OS/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/vs_globals.h:5,
                 from /data/OS/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/gfx/matrix.h:26,
                 from /data/OS/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/gfx/quaternion.h:5,
                 from /data/OS/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/opcodesysdef.h:28,
                 from /data/OS/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/Stdafx.h:21,
                 from /data/OS/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/Ice/Stdafx.h:1,
                 from /data/OS/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/Ice/IceAABB.cpp:21:
/usr/include/stdint.h:101:21: error: conflicting declaration 'typedef __intmax_t intmax_t'
  101 | typedef __intmax_t  intmax_t;

Reply:
I'd say that's a bug that needs fixing :)

Reply:
OK, I'll work on adding RelWithDebInfo back then.

Reply:
@stephengtuggy if you can repro it fine; if not, let's file an issue and we'll do a follow-up PR for it.

Reply:
This compile error is now gone.

Reply:
OK, I have added RelWithDebInfo back in, and changed CMake's default build configuration back to that. Do we want to change the default build configuration in vsbuild.sh as well?